core/src/test/scala/unit/kafka/network/SocketServerTest.scala

```diff
@@ -64,6 +66,13 @@ class SocketServerTest extends JUnitSuite {
     outgoing.flush()
   }
 
+  def sendRequest(socket: Socket, request: Array[Byte]) {
```

Comment:
Would it not make sense to change the existing `sendRequest` so that it looks something like the following?

```scala
  def sendRequest(socket: Socket, request: Array[Byte], id: Option[Short] = None) {
    val outgoing = new DataOutputStream(socket.getOutputStream)
    id match {
      case Some(id) =>
        outgoing.writeInt(request.length + 2)
        outgoing.writeShort(id)
      case None =>
        outgoing.writeInt(request.length)
    }
    outgoing.write(request)
    outgoing.flush()
  }
```

Alternatively, if we expect the other one to be removed once we transition to new request types, we should change it's name and add a comment to make that clear.